File: AppBox/ViewController/UploadAdvancedSettingViewController/UploadAdvancedSettingViewController.m
//
//  UploadAdvancedSettingViewController.m
//  AppBox
//
//  Created by Vineet Choudhary on 30/12/16.
//  Copyright © 2016 Developer Insider. All rights reserved.
//

#import "UploadAdvancedSettingViewController.h"

@implementation UploadAdvancedSettingViewController{
    
}

- (void)viewDidLoad {
    [super viewDidLoad];

    if (self.ipaUploadInfo.bundleDirectory) {
        [self.dbFolderNameTextField setStringValue:self.ipaUploadInfo.bundleDirectory.lastPathComponent];
    }
    [self.dbFolderNameTextField setEnabled:self.ipaUploadInfo.isKeepSameLinkEnabled];
}

- (void)viewDidDisappear{
    [super viewDidDisappear];
}

//MARK: - Action Button Tapped
- (IBAction)buttonCancelTapped:(NSButton *)sender {
    [self dismissController:self];
    [self.delegate uploadAdvancedSettingCancelButtonTapped:sender];
}

- (IBAction)buttonSaveTapped:(NSButton *)sender {
    if (self.delegate != nil){
        [self.delegate uploadAdvancedSettingSaveButtonTapped:sender];
    }
    
    if(![self.dbFolderNameTextField.stringValue isEqualToString:self.ipaUploadInfo.identifer] && self.dbFolderNameTextField.stringValue.length>0){
        NSString *bundlePath = [NSString stringWithFormat:@"/%@",self.dbFolderNameTextField.stringValue];
        bundlePath = [bundlePath stringByReplacingOccurrencesOfString:@" " withString:abEmptyString];
        [self.ipaUploadInfo setBundleDirectory:[NSURL URLWithString:bundlePath]];
    }

    [self dismissController:self];
}





@end

File: AppBox/ViewController/UploadAdvancedSettingViewController/UploadAdvancedSettingViewController.swift
//
//  UploadAdvancedSettingViewController.swift
//  AppBox

import AppKit

public protocol UploadAdvancedSettingViewDelegate: AnyObject {
    func uploadAdvancedSettingSaveButtonTapped(_ sender: NSButton?)
    func uploadAdvancedSettingCancelButtonTapped(_ sender: NSButton?)
}

public final class UploadAdvancedSettingViewController: NSViewController {

    public var ipaUploadInfo: IPAUploadInfo?
    public weak var delegate: UploadAdvancedSettingViewDelegate?

    private var model: AdvancedSettingsModel?

    public override func loadView() {
        let folder = ipaUploadInfo?.bundleDirectory?.lastPathComponent ?? ""
        let model = AdvancedSettingsModel(folderName: folder,
                                          fieldEnabled: ipaUploadInfo?.isKeepSameLinkEnabled ?? false)
        self.model = model
        model.onSave = { [weak self] in self?.saveSettings() }
        model.onCancel = { [weak self] in self?.cancelSettings() }
        view = AdvancedSettingsHost.makeView(model: model)
    }

    // MARK: - Actions (invoked by the SwiftUI view's callbacks)

    private func cancelSettings() {
        dismiss(self)
        delegate?.uploadAdvancedSettingCancelButtonTapped(nil)
    }

    private func saveSettings() {
        delegate?.uploadAdvancedSettingSaveButtonTapped(nil)

        if let folder = model?.folderNameText, folder != ipaUploadInfo?.identifer, !folder.isEmpty {
            let bundlePath = "/\(folder)".replacingOccurrences(of: " ", with: "")
            ipaUploadInfo?.bundleDirectory = URL(string: bundlePath)
        }

        dismiss(self)
    }
}
